lang/java/reef-runtime-multi/src/main/java/org/apache/reef/runtime/multi/client/AbstractDriverConfigurationProvider.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runtime.multi.client;
+
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.io.EncoderFactory;
+import org.apache.avro.io.JsonEncoder;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.apache.reef.runtime.common.client.DriverConfigurationProvider;
+import org.apache.reef.runtime.multi.driver.MultiRuntimeDriverConfiguration;
+import org.apache.reef.runtime.multi.utils.RuntimeDefinition;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.Configurations;
+import org.apache.reef.tang.formats.AvroConfigurationSerializer;
+import org.apache.reef.tang.formats.ConfigurationModule;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+
+/**
+ * Provides base class for driver configuration providers for multi runtimes.
+ */
+abstract class AbstractDriverConfigurationProvider implements DriverConfigurationProvider {
+
+  private static final String CHARSET_NAME = "ISO-8859-1";
+
+  protected static String serializeConfiguration(final ConfigurationModule configModule) {
+    final Configuration localDriverConfiguration = configModule.build();
+    final AvroConfigurationSerializer serializer = new AvroConfigurationSerializer();
+    return serializer.toString(localDriverConfiguration);
+  }
+
+  protected static String serializeRuntimeDefinition(final String serializedConfig,
+                                                     final boolean defaultRuntime,
+                                                     final String runtimeName) {
+    RuntimeDefinition rd =
```

Comment:
`final`. IntelliJ has an inspector that can find all cases of variables that can be, but are not, `final`. Please apply that to this change.